### PR TITLE
fix: add a callback function on flush_rows

### PR DESCRIPTION
**Pull request by mutianf**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigtable/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: google/cloud/bigtable/batcher.py [hunk 1]
```diff
@@ -192,6 +192,10 @@ class MutationsBatcher(object):
     :type flush_interval: float
     :param flush_interval: (Optional) The interval (in seconds) between asynchronous flush.
         Default is 1 second.
+
+    :type flush_completed_callback: Callable = None
+    :param flush_completed_callback: (Optional) A callable funtion for handling responses
```
Comment (igorbernstein2) [inline google/cloud/bigtable/batcher.py]: I dont think flush is correct here...from a user perspective it means all buffered elements have been sent, but that might not be the case. I think it should be batch_completed_callback\

File: google/cloud/bigtable/batcher.py [hunk 2]
```diff
@@ -215,6 +220,7 @@ def __init__(
         )
         self.futures_mapping = {}
         self.exceptions = queue.Queue()
+        self.flush_completed_callback = flush_completed_callback
```
Comment (igorbernstein2) [inline google/cloud/bigtable/batcher.py:223]: please prefix with _ to mark it as provite

File: google/cloud/bigtable/batcher.py [hunk 3]
```diff
@@ -357,6 +363,9 @@ def _flush_rows(self, rows_to_flush):
         if len(rows_to_flush) > 0:
             response = self.table.mutate_rows(rows_to_flush)
 
+            if self.flush_completed_callback:
+                self.flush(response)
```
Comment (igorbernstein2) [inline google/cloud/bigtable/batcher.py:367]: this look very wrong :)

**Review by igorbernstein2 — CHANGES_REQUESTED** [review]

File: google/cloud/bigtable/batcher.py (hunk 3 above)
Comment (mutianf) [reply google/cloud/bigtable/batcher.py:367]: oops 😂

**Review by igorbernstein2 — APPROVED** [review]

**Review by igorbernstein2 — APPROVED** [review]

**Review by daniel-sanche — APPROVED** [review]

**Pull request merged**